pontoon/actionlog/migrations/0004_auto_20240816_1045.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 4.2.11 on 2024-08-16 10:45
+
+import datetime
+
+from django.db import migrations
+
+
+def migrate_translation_to_actionlog(apps, schema_editor):
+    Translation = apps.get_model("base", "Translation")
+    ActionLog = apps.get_model("actionlog", "ActionLog")
+
+    translation_info = {
+        "date": ("translation:created", "Translation created"),
+        "approved_date": ("translation:approved", "Translation approved"),
+        "unapproved_date": ("translation:unapproved", "Translation unapproved"),
+        "rejected_date": ("translation:rejected", "Translation rejected"),
+        "unrejected_date": ("translation:unrejected", "Translation unrejected"),
+    }
+
+    # date to end migration
+    end_date = datetime.datetime(
+        2020, 1, 7, 9, 25, 11, 829125, tzinfo=datetime.timezone.utc
+    )
+
+    actions_to_log = []
+
+    for translation in Translation.objects.filter(date__lt=end_date):
+        for attr, action_type in translation_info.items():
+            value = getattr(translation, attr)
+            if value is not None and value < end_date:
+                actions_to_log.append(
+                    ActionLog(
+                        action_type=action_type,
+                        created_at=value,
+                        performed_by_id=translation.user_id,
```

Comment:
Sorry, I missed one more thing - `performed_by` should be a different value for different values of `attr`:
* `date`: `translation.user`
* `approved_date`: `translation.approved_user`
* `unapproved_date`: `translation.unapproved_user`
* `rejected_date`: `translation.rejected_user`
* `unrejected_date`: `translation.unrejected_user`

Additionally, we should only store and entry to the `ActionLog` if the said user value is not `None`.